internal/services/netapp/validate/volume_group_sap_hana_volumes_validation_test.go

```diff
@@ -0,0 +1,350 @@
+					},
+				},
+				{ // log
+					Name: pointer.To(fmt.Sprintf("volume-%v", string(VolumeSpecNameSapHanaLog))),
+					Properties: volumegroups.VolumeProperties{
+						ExportPolicy: &volumegroups.VolumePropertiesExportPolicy{
+							Rules: &[]volumegroups.ExportPolicyRule{
+								{
+									Nfsv3:  pointer.To(false),
+									Nfsv41: utils.Bool(true),
+								},
+							},
+						},
+						ProtocolTypes:           pointer.To([]string{"NFSv4.1"}),
+						ProximityPlacementGroup: pointer.To("/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Compute/proximityPlacementGroups/ppg1"),
+						SecurityStyle:           pointer.To(volumegroups.SecurityStyle(SecurityStyleUnix)),
+						VolumeSpecName:          pointer.To(string(VolumeSpecNameSapHanaLog)),
+					},
+				},
+				{ // shared
+					Name: pointer.To(fmt.Sprintf("volume-%v", string(VolumeSpecNameSapHanaShared))),
+					Properties: volumegroups.VolumeProperties{
+						ExportPolicy: &volumegroups.VolumePropertiesExportPolicy{
+							Rules: &[]volumegroups.ExportPolicyRule{
+								{
+									Nfsv3:  pointer.To(false),
+									Nfsv41: utils.Bool(true),
+								},
+							},
+						},
+						ProtocolTypes:           pointer.To([]string{"NFSv4.1"}),
+						ProximityPlacementGroup: pointer.To("/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Compute/proximityPlacementGroups/ppg1"),
+						SecurityStyle:           pointer.To(volumegroups.SecurityStyle(SecurityStyleUnix)),
+						VolumeSpecName:          pointer.To(string(VolumeSpecNameSapHanaShared)),
+					},
+				},
+				{ // data-backup
+					Name: pointer.To(fmt.Sprintf("volume-%v", string(VolumeSpecNameSapHanaDataBackup))),
+					Properties: volumegroups.VolumeProperties{
+						ExportPolicy: &volumegroups.VolumePropertiesExportPolicy{
+							Rules: &[]volumegroups.ExportPolicyRule{
+								{
+									Nfsv3:  pointer.To(false),
+									Nfsv41: utils.Bool(true),
+								},
+							},
+						},
+						ProtocolTypes:  pointer.To([]string{"NFSv4.1"}),
+						SecurityStyle:  pointer.To(volumegroups.SecurityStyle(SecurityStyleUnix)),
+						VolumeSpecName: pointer.To(string(VolumeSpecNameSapHanaDataBackup)),
+					},
+				},
+				{ // log-backup
+					Name: pointer.To(fmt.Sprintf("volume-%v", string(VolumeSpecNameSapHanaLogBackup))),
+					Properties: volumegroups.VolumeProperties{
+						ExportPolicy: &volumegroups.VolumePropertiesExportPolicy{
+							Rules: &[]volumegroups.ExportPolicyRule{
+								{
+									Nfsv3:  pointer.To(false),
+									Nfsv41: utils.Bool(true),
+								},
+							},
+						},
+						ProtocolTypes:  pointer.To([]string{"NFSv4.1"}),
+						SecurityStyle:  pointer.To(volumegroups.SecurityStyle(SecurityStyleUnix)),
+						VolumeSpecName: pointer.To(string(VolumeSpecNameSapHanaLogBackup)),
+					},
+				},
+			},
+			Errors: 0,
+		},
+		{
+			Name: "ValidateMinimumVolumes",
```

Comment:
We need a test that Errors when you don't have the minimum number of Volumes

Reply:
Done, new test called `ValidateLessThanMinimumVolumes`

```
=== RUN   TestValidateNetAppVolumeGroupSAPHanaVolumes
=== RUN   TestValidateNetAppVolumeGroupSAPHanaVolumes/ValidateCorrectSettingsAllVolumes
=== RUN   TestValidateNetAppVolumeGroupSAPHanaVolumes/ValidateMinimumVolumes
=== RUN   TestValidateNetAppVolumeGroupSAPHanaVolumes/ValidateRequiredVolumeSpecs
=== RUN   TestValidateNetAppVolumeGroupSAPHanaVolumes/ValidateLessThanMinimumVolumes
=== RUN   TestValidateNetAppVolumeGroupSAPHanaVolumes/ValidateNoNfsVersionThreeOnDataLogAndSharedVolumes
=== RUN   TestValidateNetAppVolumeGroupSAPHanaVolumes/ValidateNoPPGBackupVolumes
=== RUN   TestValidateNetAppVolumeGroupSAPHanaVolumes/ValidateRequiredPpgForNonBackupVolumes
=== RUN   TestValidateNetAppVolumeGroupSAPHanaVolumes/ValidateVolumeSpecCantRepeat
=== RUN   TestValidateNetAppVolumeGroupSAPHanaVolumes/ValidateEndpointDstNotEnabledOnLogVolume
=== RUN   TestValidateNetAppVolumeGroupSAPHanaVolumes/ValidateSnapshotPolicyNotEnabledOnEndpointDstVolume
--- PASS: TestValidateNetAppVolumeGroupSAPHanaVolumes (0.00s)
    --- PASS: TestValidateNetAppVolumeGroupSAPHanaVolumes/ValidateCorrectSettingsAllVolumes (0.00s)
    --- PASS: TestValidateNetAppVolumeGroupSAPHanaVolumes/ValidateMinimumVolumes (0.00s)
    --- PASS: TestValidateNetAppVolumeGroupSAPHanaVolumes/ValidateRequiredVolumeSpecs (0.00s)
    --- PASS: TestValidateNetAppVolumeGroupSAPHanaVolumes/ValidateLessThanMinimumVolumes (0.00s)
    --- PASS: TestValidateNetAppVolumeGroupSAPHanaVolumes/ValidateNoNfsVersionThreeOnDataLogAndSharedVolumes (0.00s)
    --- PASS: TestValidateNetAppVolumeGroupSAPHanaVolumes/ValidateNoPPGBackupVolumes (0.00s)
    --- PASS: TestValidateNetAppVolumeGroupSAPHanaVolumes/ValidateRequiredPpgForNonBackupVolumes (0.00s)
    --- PASS: TestValidateNetAppVolumeGroupSAPHanaVolumes/ValidateVolumeSpecCantRepeat (0.00s)
    --- PASS: TestValidateNetAppVolumeGroupSAPHanaVolumes/ValidateEndpointDstNotEnabledOnLogVolume (0.00s)
    --- PASS: TestValidateNetAppVolumeGroupSAPHanaVolumes/ValidateSnapshotPolicyNotEnabledOnEndpointDstVolume (0.00s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/netapp/validate       0.015s
```